### ISSUE TITLE
Any forks on this?

### ISSUE THREAD

**Issue by trymeouteh** [syncthing/syncthing-gtk#591]
This has not been maintained in a few years but is a good GUI for Syncthing on Linux.

Are there any forks of it out there that anyone know of?

**Comment by hrehfeld** [comment]
I took the liberty to open an organization for this, maybe we can maintain a fork there, even if temporary? https://github.com/syncthing-gtk/syncthing-gtk @Rudd-O is invited as an owner.

**Comment by hrehfeld** [comment]
Ah, but there is also https://salsa.debian.org/debian/syncthing-gtk, no idea how related/changed it is.

**Comment by Rudd-O** [comment]
Thanks @hrehfeld I think if @kozec is not responding, we should simply use that org.  @hrehfeld feel free to merge my pull requests onto your fork of this thing.

**Comment by hrehfeld** [comment]
> Thanks @hrehfeld I think if @kozec is not responding, we should simply use that org. @hrehfeld feel free to merge my pull requests onto your fork of this thing.

:+1: 

I directly forked your repo, so everything should be there. Also, you should have an invite as an owner.

Maybe we should ping the debian guys and see what they think?

**Comment by tpikonen** [comment]
I have a cleaned up tree of syncthing-gtk [here](https://github.com/tpikonen/syncthing-gtk/tree/pep8). It's a merge of the version in Debian by @andrewshadura and the @Rudd-O branch. It's flake8 (pep8 + pyflakes) clean and also has a couple of small fixes on top of the code cleanup.

Feel free to build on top it, cleaning the code took a couple of evenings, so you'll save yourself that. I have some further plans with the code, but don't have time for them right now.

**Comment by andrewshadura** [comment]
Oh, I’m a bit surprised this all happened without anyone telling me 😕

**Comment by andrewshadura** [comment]
I’m not impressed by this situation. My fork was created ages ago, I gradually worked on these issues, only for my work to be partially re-done and lost.

**Comment by hrehfeld** [comment]
What do you suggest how we proceed under the assumption that the original maintainer is gone at least temporary?

19 Sep 2022 23:19:25 Andrej Shadura ***@***.***>:

**Comment by andrewshadura** [comment]
Well, I tried to maintain a working fork and I actually started some refactoring and UX rework (according to #567), but true, I didn’t work on it for a couple of months as I was busy with other work. So at least I’d expect that Manuel would take my work as a base for his Fedora port and contribute back, and maybe that could become a new upstream for Syncthing GTK. As you may have noticed, I tried to keep the history clean and polish up all patches I took from other forks.

**Comment by andrewshadura** [comment]
By the way, my assumption is that the original author is not coming back, and, honestly, I’d not be surprised if this repo would be gone any day with all reported issues and other things.

**Comment by Rudd-O** [comment]
Let's continue with the syncthing forked org and move on from this maintainer.  I will be working with that forked branch (which included my changes) from now on.

**Comment by andrewshadura** [comment]
@Rudd-O, given that you re-implemented most of the work I did in my branch, wouldn’t it make sense to start with mine instead?

**Comment by Rudd-O** [comment]
https://github.com/syncthing-gtk/syncthing-gtk already has everything plus more fixes than my work alone, and it works fine.  Please merge your improvements there!  Thank you for your contributions!  :-)

**Comment by andrewshadura** [comment]
Does it? This seems very unfair, having my work thrown out despite it being around for ages.

**Comment by Rudd-O** [comment]
I think I used your work as a base for mine!  Either way both our works would inevitably converge on the same result — pep8 blackened code that works with Python 3.

**Comment by andrewshadura** [comment]
You didn’t, otherwise you wouldn’t need to port it to Python 3 once again. In any case, porting to Python 3 wasn’t the only thing I did.

**Comment by Rudd-O** [comment]
Correct, which is why your work is definitely beneficial for the now-active fork of the project.

**Comment by hrehfeld** [comment]
the aur package for syncthing-gtk-python3 is now pointing to https://github.com/Salamandar/syncthing-gtk . Yet another fork? @Salamandar

**Comment by andrewshadura** [comment]
It's my fork with an extra patch, apparently.

**Comment by hrehfeld** [comment]
@andrewshadura I get your grief, but please don't feel insulted. I simply didn't find your fork earlier (and it seems @Rudd-O did neither); shit happens. I'd really appreciate it if we get some traction maintaining this together, and I don't think work is lost, really.

**Comment by Salamandar** [comment]
@hrehfeld @andrewshadura Yes I forked it from salsa because it need one patch for the AUR package, and salsa didn't accept my register request (without explanation), so I couldn't even do a MR on it…

@hrehfeld doing an org on github is a good idea. I'm willing to delete my fork if I'm added to it :D (the repo can have protected master branch, with PR-only merges).

I was actually working on cleaning up the code. There still is a lot of stuff from the python2 era. Also, pushing for gtk4 might be something worth working on.

**Comment by andrewshadura** [comment]
I was also working on clean-ups, I tried porting to GTK 4, but I was first focusing on getting rid of custom widgets in favour of standard GTK ones. Unfortunately, I wasn’t able to spend much time on it lately.

**Comment by Salamandar** [comment]
Yes, let's work on it together instead of each on its own repo ;)

**Comment by andrewshadura** [comment]
@hrehfeld, I’ve just compared my fork and the other one, and apart from the fact the history is much cleaner in my version, many important fixes are missing in the one Rudd-O created.

I propose this way forward: I’ll spend some time this week to reconcile the forks and merge Rudd-O’s changes into my more complete fork, then we push it as a `main` branch into the new repo and make `main` the default branch. I also have a bunch of work in progress things which I’ll also push there.

**Comment by Rudd-O** [comment]
Why not work in the Syncthing-GTK organization that Salamandar created?

**Comment by Salamandar** [comment]
> Why not work in the Syncthing-GTK organization that Salamandar created?

You mean @hrehfeld I think :) But yes, I agree !

**Comment by Salamandar** [comment]
As it looks like the choice of the fork is subject to debate, a "politically correct" way to start this fork would be to take this original repo as base and everyone would make their own pull request on it :p 
That would be a bit of work to redo the PRs, but at least it wouldn't favorise anyone's work…

**Comment by Rudd-O** [comment]
My main priority is less favoritism — rather it is to deliver the biggest and most impactful fixes and improvements to the thing that then everyone uses.  That means this work would have to be ego-less.  An organization seems to me to be the right way to go — even if my code is wholesale thrown out — of course, so long as whatever is published through the org is, in fact, functional.

**Comment by Rudd-O** [comment]
Furthermore, instead of starting with a monstrous PR, I think it's best if we receive small incremental PRs that fix things as independently as possible, so that we may review them and test them more easily.  My 2 cents.

**Comment by andrewshadura** [comment]
Yes, I meant the organisation was the way to go.

**Comment by Rudd-O** [comment]
@hrehfeld can you enable the issue tracker in the org's repo?  It does not appear onscreen to me.

EDIT: d'oh, I could do it.  Did it.  Hah.

**Comment by Rudd-O** [comment]
Who wants added to the org repo?  You'd be receiving PR review requests if that's alright with you.

**Comment by Salamandar** [comment]
> Who wants added to the org repo? You'd be receiving PR review requests if that's alright with you.

I'd love to. Can't promise to respond to every MR but i'll try to do as much as possible.

**Comment by hrehfeld** [comment]
> 

Issues should be enabled: https://github.com/syncthing-gtk/syncthing-gtk/issues

**Comment by hrehfeld** [comment]
I invited all of you. You should have an invite for the syncthing-gtk team as well as the repo. Probably repo is unnecessary, but well :shrug:

**Comment by hrehfeld** [comment]
I'm fine with either way of starting/maintaining/choice of branch, as long as we work together :heart:

**Comment by andrewshadura** [comment]
I’ve just found [this comment](https://news.ycombinator.com/item?id=33598867) — and indeed, the new repo in the syncthing-gtk orga cannot be found. Should we re-create the repo as a non-fork for better discoverability?

**Comment by Rudd-O** [comment]
I think we should preserve history for now.  It's the 6th result in Google now.

![image](https://user-images.githubusercontent.com/346306/201876721-ddc9acaf-f0fd-4fc3-9513-7068ce5e00b1.png)

**Comment by Rudd-O** [comment]
Eventually we can figure out if things need to be flipped.

**Comment by Salamandar** [comment]
@andrewshadura @Rudd-O  see https://github.com/syncthing-gtk/syncthing-gtk/issues/3 :)

**Comment by Rudd-O** [comment]
We should probably continue our conversation in that ticket.
